### PR TITLE
Build PAPI@7 with CUDA support

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

File: P/PAPI/build_tarballs.jl [hunk 1]
```diff
@@ -42,9 +56,43 @@ products = [
...
+            BuildDependency(PackageSpec(name="CUDA_full_jll",
+                                        version=cuda_full_versions[cuda_version])),
+            RuntimeDependency(PackageSpec(name="CUDA_Runtime_jll")),
+        ]
+    end
+
+    build_tarballs(ARGS, name, version, sources, script, [augmented_platform],
+                   products, dependencies; lazy_artifacts=true,
+                   julia_compat="1.6", augment_platform_block)
+end
```
Comment (giordano) [inline P/PAPI/build_tarballs.jl:98]:
```suggestion
end

```

File: P/PAPI/build_tarballs.jl [hunk 2]
```diff
@@ -42,9 +56,43 @@ products = [
...
+
+    if cuda_version == "none"
+        dependencies = []
+    else
+        if arch(platform)
+        dependencies = [
+            BuildDependency(PackageSpec(name="CUDA_full_jll",
+                                        version=cuda_full_versions[cuda_version])),
+            RuntimeDependency(PackageSpec(name="CUDA_Runtime_jll")),
+        ]
```
Comment (giordano) [inline P/PAPI/build_tarballs.jl:92]: I think this should be
```suggestion
            dependencies = [
                BuildDependency(PackageSpec(name="CUDA_full_jll",
                                            version=cuda_full_versions[cuda_version])),
                RuntimeDependency(PackageSpec(name="CUDA_Runtime_jll")),
            ]
        end
```
?

**Comment by vchuravy** [comment]
Ah I forgotten to save...

File: P/PAPI/build_tarballs.jl [hunk 3]
```diff
@@ -1,28 +1,70 @@
...
+
+export PAPI_CUDA_ROOT="${prefix}/cuda"
+COMPONENTS=()
+if [[ -d "${prefix}/cuda" ]]; then
+    COMPONENTS+=(cuda)
+fi
+
+if [[ "${target} == "powerpc64le-*"" ]]; then
+  CPU=POWER8
+elif [[ "${target}" == "x86_64-*" || "${target}" == "i686-*" ]]; then
```
Comment (giordano) [inline P/PAPI/build_tarballs.jl:43]: You can also use
```suggestion
elif [[ "${proc_family}" == "intel" ]]; then
```
see https://docs.binarybuilder.org/stable/build_tips/#Automatic-environment-variables.  Also, I'm not sure the `*` wildcard plays well with quoting there.

File: P/PAPI/build_tarballs.jl [hunk 4]
```diff
@@ -1,28 +1,70 @@
...
     CFLAGS="-D_GNU_SOURCE"
 fi
+
+export PAPI_CUDA_ROOT="${prefix}/cuda"
+COMPONENTS=()
+if [[ -d "${prefix}/cuda" ]]; then
+    COMPONENTS+=(cuda)
+fi
+
+if [[ "${target} == "powerpc64le-*"" ]]; then
```
Comment (vchuravy) [inline P/PAPI/build_tarballs.jl:41]:
```suggestion
if [[ "${target} == "powerpc64le-*" ]]; then
```

File: P/PAPI/build_tarballs.jl [hunk 5]
```diff
@@ -1,28 +1,70 @@
...
     CFLAGS="-D_GNU_SOURCE"
 fi
+
+export PAPI_CUDA_ROOT="${prefix}/cuda"
+COMPONENTS=()
+if [[ -d "${prefix}/cuda" ]]; then
+    COMPONENTS+=(cuda)
+fi
+
+if [[ "${target} == "powerpc64le-*" ]]; then
```
Comment (vchuravy) [inline P/PAPI/build_tarballs.jl:41]:
```suggestion
if [[ "${target}" == "powerpc64le-*" ]]; then
```

**Comment by giordano** [comment]
Good to go?

**Pull request merged**